cli/tests/delayed_error.ts

```diff
@@ -0,0 +1,2 @@
+await new Promise((r) => setTimeout(r, 100));
```

Comment:
Can this line be replaced with `await Promise.resolve();`? If so maybe you could rewrite it as a unit test in `core/runtime.rs`?

Reply:
It must be delayed by a macrotask at least.